src/System.Net.Http/tests/UnitTests/HttpEnvironmentProxyTest.cs

```diff
@@ -204,7 +205,22 @@ public void HttpProxy_Exceptions_Match()
                 Assert.True(p.IsBypassed(new Uri("http://www.test.com")));
 
                 return SuccessExitCode;
-           }).Dispose();
+            }).Dispose();
+        }
+
+        [Fact]
+        public void HttpProxy_TryCreate_UpperCaseVariables()
```

Comment:
This doesn't seem to be validating the HTTP_PROXY. Also, could we generalize this test a bit more to help validate both the lower and upper cases? e.g. something like (untested):
```C#
[Theory]
[InlineData("HTTP_PROXY", "NO_PROXY")]
[InlineData("http_proxy", "no_proxy")]
public void HttpProxy_TryCreate_UpperCaseVariables(string httpProxy, string noProxy)
{
    var options = new RemoteInvokeOptions();
    options.ProcessStartInfo.Environment.Add(httpProxy, "http://foo:bar@1.1.1.1:3000");
    options.ProcessStartInfo.Environment.Add(noProxy, ".test.com, foo.com");
    RemoteInvoke(() =>
    {
        Assert.True(HttpEnvironmentProxy.TryCreate(out IWebProxy p));
        Assert.NotNull(p);

        Assert.True(p.IsBypassed(new Uri("http://test.com")));
        Assert.Null(p.GetProxy(new Uri("http://test.com")));

        Assert.False(p.IsBypassed(new Uri("http://1test.com")));
        Assert.Equal("http://foo:bar@1.1.1.1:3000", p.GetProxy(new Uri("http://1test.com").AbsoluteUri);

        return SuccessExitCode;
    }, options).Dispose();
}
```

Reply:
Yes i will add this. Anyway Null check for get proxy will not be returned since there is currently no check for NO_PROXY in this method.

Do you want me to add this check also inside GetProxy method ?

Reply:
I pushed the changed code without this GetProxy change. Maybe this should be a part of other pull request if it should be changed